scopes/webpack/modules/fragments/style-preset/style-preset.ts

```diff
@@ -0,0 +1,202 @@
+import type { RuleSetRule, WebpackPluginInstance } from 'webpack';
+import type { Options as SassLoaderOptions } from 'sass-loader';
+import MiniCssExtractPlugin, { PluginOptions } from 'mini-css-extract-plugin';
+import getCSSModuleLocalIdent from 'react-dev-utils/getCSSModuleLocalIdent';
+import { allCssRegex, cssRegex, sassRegex, lessRegex } from '@teambit/webpack.modules.style-regexps';
+
+const styleLoaderPath = require.resolve('style-loader');
+// * all deps are rather small (2kb - 50kb) - so it's ok to import both style-loader and miniCss
+// * no need for 'sourceMaps' - loaders respect `devTools` (preferably, 'inline-source-map')
+// * miniCssExtractPlugin - (somehow) breaks hot reloading, so using style-loader in dev mode
+
+export type WebpackLoader = string;
+
+export type CssOptions = {
+  /**
+   * type of style injecctor to use.
+   * style-loader has better hot reloading support but is inline,
+   * mini-css-extract-plugin actually creates css files
+   */
+  styleInjector?: 'mini-css-extract-plugin' | 'style-loader';
+  /**
+   * handle *.module.* as css-modules
+   * @default true
+   */
+  modules?: boolean | RegExp | ((filename: string) => boolean);
+
+  /** override specific path to css-loader */
+  cssLoader?: WebpackLoader;
+  /** loader name or path, override the default postcss loader */
+  postcssLoader?: WebpackLoader;
+
+  /** options for MiniCssExtractPlugin */
+  miniCssOptions?: PluginOptions;
+  /** override the options of css-loader */
+  cssLoaderOptions?: any;
+  /** override css injector options */
+  injectorOptions?: any;
+  /** options for postcss-loader (loader is skipped when undefined) */
+  postcssOptions?: any;
+};
+
+export type CssDialectOptions = {
+  /** loader name or path, override the default resolve-url-loader */
+  resolveUrlLoader?: WebpackLoader;
+  /** loader name or path, override the default sass-loader */
+  sassLoader?: WebpackLoader;
+  /** loader name or path, override the default less-loader */
+  lessLoader?: WebpackLoader;
+  /** override the options for sass loader */
+  sassLoaderOptions?: SassLoaderOptions;
+  /** override the options for less loader */
+  lessLoaderOptions?: any;
+  /** override the options for resolveUrl loader */
+  resolveUrlOptions?: any;
+};
+
+export type StyleLoadersOptions = CssOptions & CssDialectOptions;
+
+/** loaders for css, including dialects (sass and less), postcss, and css-modules */
+export const makeStyleLoaders = ({
```

Comment:
let's rename it to generate instead of make?

Reply:
done